### DISCUSSION TITLE
With Portable Mode VSCode, where does the .ipch file get stored?

### DISCUSSION THREAD

**Discussion by TryerGit** [microsoft/vscode-cpptools#11681] [Q&A] [answered]
With portable mode VSCode, https://code.visualstudio.com/docs/editor/portable, and installing cpptools extension, where do the .ipch file and other cpptools generated files get stored? Current official documentation, https://code.visualstudio.com/docs/cpp/faq-cpp#_what-is-the-ipch-folder, does not clarify this. It indicates where the temp files/folders will be generated (I assume) only in the "full" (as opposed to portable mode) VSCode installation, for e.g., `%LocalAppData%/Microsoft/vscode-cpptools` in Windows and `$HOME/.cache/vscode-cpptools/` in Linux. 

Is it guaranteed that under portable mode VSCode, all generated cpptools files and folders (such as .ipch files/folders) will go into the portable mode VSCode installed folder only?

**Comment by sean-mcmanus** [comment, marked as answer]
@TryerGit I don't know. We haven't tested with portable mode VS Code or written any special case code to handle that scenario. Can you try it out? You can change the C_Cpp.intelliSenseCachePath and the C_Cpp.default.browse.databaseFilename.

**Reply by TryerGit** [reply]
Fair enough @sean-mcmanus  I will try it out and update this thread in due course. One of my chief concerns is to ensure that VSCode does not end up creating too many files that I am not aware of. Could you please confirm if the ipch file and the browse.db files are the only "temp" files created by the cpptools extension that I should look out for and see where they are created? If there are other files, and their location is also configurable, please do let me know. I will watch out for them and configure them so that they can go inside the portable mode folder also. Thank you.

**Reply by sean-mcmanus** [reply]
@TryerGit Ah, yeah, our extension could be creating other temp files in certain scenarios, but those are typically smaller in size and would go to some OS dependent temp folder and we don't have a setting to configure those files. The ipch/browse.db are the only files that are expected to be large in size.

**Comment by starball5** [comment]
Related on Stack Overflow: [How does VS Code portable mode interact with extensions and files generated by the extensions?](https://stackoverflow.com/q/77481953/11107541)

**Comment by MatejKafka** [comment]
Cpptools is the only VS Code extension I know about that stores data outside the vscode data directory, which is especially unfortunate when VS Code is running in portable mode. Is there any chance you could move the cache directory from `AppData/Local` to the VS Code data directory?

This is compounded by the fact that there's no simple way to actually move the whole cache directory:
1) There's no way to move the database files while preserving the directory structure of each database file being in a separate directory with hashed name.
2) Even when both the ipch dir and the database directories are moved using `C_Cpp.default.browse.databaseFilename` and `C_Cpp.intelliSenseCachePath`, the extension still creates empty directories under `AppData/Local/Microsoft/vscode-cpptools`.
